pkgs/development/libraries/precice/default.nix

```diff
@@ -12,14 +12,21 @@ stdenv.mkDerivation rec {
   };
 
   preConfigure = ''
-    cmakeFlags="-DBUILD_SHARED_LIBS=ON -DPETSC=off"
+    cmakeFlags="-DBUILD_SHARED_LIBS=ON
+                -DPETSC=off
+                -DCMAKE_BUILD_TYPE=Debug
```

Comment:
This too, if you use `cmakeBuildType = "Debug"` attribute. Though there should not be a need for this unless you actually enable debugging with `enableDebugging` since Nix will strip the debug symbols otherwise.